pennylane/pulse/transmon.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains the classes/functions specific for simulation of superconducting transmon hardware systems"""
+from dataclasses import dataclass
+from typing import Callable, List, Union
+
+import pennylane as qml
+from pennylane.pulse import HardwareHamiltonian
+from pennylane.typing import TensorLike
+
+
+# TODO ladder operators once there is qudit support
+# pylint: disable=unused-argument
+def a(wire, d=2):
+    """creation operator"""
+    return qml.s_prod(0.5, qml.PauliX(wire)) + qml.s_prod(0.5j, qml.PauliY(wire))
+
+
+def ad(wire, d=2):
+    """annihilation operator"""
+    return qml.s_prod(0.5, qml.PauliX(wire)) + qml.s_prod(-0.5j, qml.PauliY(wire))
+
+
+# pylint: disable=too-many-arguments
+def transmon_interaction(
+    omega: Union[float, list],
+    connections: list,
+    g: Union[float, list],
+    anharmonicity=None,
+    wires=None,
+    d=2,
+):
+    r"""Returns a :class:`ParametrizedHamiltonian` representing the cQED Hamiltonian of a superconducting transmon system.
+
+    The Hamiltonian is given by
+
+    .. math::
+
+        H = \sum__{q\in \text{wires}} \omega_q a^\dagger_q a_q
+        + \sum_{(i, j) \in \mathcal{C}} g_{ij} \left(a^\dagger_i a_j + a_j^\dagger a_i \right)
+        + \sum_{q\in \text{wires}} \alpha_q a^\dagger_q a^\dagger_q a_q a_q
+
+    where :math:`[a^\dagger_p, a_q] = i \delta_{pq}` are bosonic creation and annihilation operators.
+    The first term describes the dressed qubit frequencies :math:`\omega_q`, the second term their
+    coupling :math:`g_{ij}` and the last the anharmonicity :math:`\alpha_q`, which all can vary for
+    different qubits. In practice, the bosonic operators are restricted to a finite dimension of the
+    local Hilbert space (default ``d=2`` corresponds to qubits).
+    In that case, the anharmonicity is set to :math:`\alpha=0` and ignored.
+
+    The values of :math:`\omega \pm \alpha` are typically around :math:`(5 \pm 0.3) \times 2\pi \text{GHz}` and it is common
```

Comment:
"`(5 \pm 0.3) \times 2\pi \text{GHz}`"

Including the "\pm 0.3" makes this feel super specific - I'm not sure about specifying with a variance like that, I don't think its so standard that we should specify a narrow range. I would remove the \pm, I think it implies more certainty/uniformity in systems than is realistic.

Also where is this number from? I've measured transmons ranging in drive frequency from 4-ish to 6-ish GHz, so I'm thinking maybe the factor 2pi doesn't belong there? The RF sources I've seen for signal generation usually max out between 10 at 20 GHz, and the qubits are fabricated to function within hardware specs. 

Coupling strength also sounds more reasonable to me without the 2pi factor, I think our 2-qubit coupling was usually on the order of 10s of MHz.

Reply:
Just to be sure, this was supposed to be compact formulation for $\omega \pm \alpha = (5 \pm 0.3) \times 2\pi \text{GHz}$, i.e. that omega is around 5 and alpha around 0.3

Reply:
As to the values themselves, they are taken from the references below, i.e. table I https://arxiv.org/pdf/2203.06818.pdf

Reply:
Hahaha, right. I see it now 🙈 

> they are taken from the references below

Ah, okay, thanks! I guess it looks like I'm thinking of the output frequency of the control hardware ($f$), and the value $\omega$ used in calculating the hamiltonian is angular frequency. Something to remember when converting to hardware, I think we will have to remove the factor of $2 \pi$ there.